gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/TraversalStrategies.java

```diff
@@ -69,10 +70,15 @@
     static List<Class<? extends TraversalStrategy>> STRATEGY_CATEGORIES = Collections.unmodifiableList(Arrays.asList(TraversalStrategy.DecorationStrategy.class, TraversalStrategy.OptimizationStrategy.class, TraversalStrategy.ProviderOptimizationStrategy.class, TraversalStrategy.FinalizationStrategy.class, TraversalStrategy.VerificationStrategy.class));
 
     /**
-     * Return all the {@link TraversalStrategy} singleton instances associated with this {@link TraversalStrategies}.
+     * Return all the {@link TraversalStrategy} instances associated with this {@link TraversalStrategies}.
      */
     public List<TraversalStrategy<?>> toList();
 
+    /**
+     * Return all the {@link TraversalStrategy} instances associated with this {@link TraversalStrategies}.
+     */
+    public Iterator<TraversalStrategy<?>> toIterator();
```

Comment:
I think this is a sign that `TraversalStrategies` should extend `Iterable<TraversalStrategy<?>>` instead.

Reply:
I thought about that, but was trying not to do too much. I created an issue for further consideration: https://issues.apache.org/jira/browse/TINKERPOP-2311